gadgets/trace_tcp/ebpf/connect.h

```diff
@@ -0,0 +1,151 @@
+// SPDX-License-Identifier: GPL-2.0
+// Copyright (c) 2024 Microsoft Corporation
+
+#ifndef __IG_TCP_CONNECT_H
+#define __IG_TCP_CONNECT_H
+
+#include "common.h"
+#include <linux/errno.h>
+
+/*
+ * tcp_set_state doesn't run in the context of the process that initiated the
+ * connection so we need to store a map TUPLE -> PID to send the right PID on
+ * the event.
+ */
+struct {
+	__uint(type, BPF_MAP_TYPE_HASH);
+	__uint(max_entries, MAX_ENTRIES);
+	__type(key, struct tuple_key_t);
+	__type(value, struct gadget_process);
+} tuplepid SEC(".maps");
+
+/*
+ * We store the socket pointer in a map to be able to retrieve it in the
+ * kretprobe.
+ */
+struct {
+	__uint(type, BPF_MAP_TYPE_HASH);
+	__uint(max_entries, MAX_ENTRIES);
+	__type(key, u32); // tid
+	__type(value, struct sock *);
+} tcp_connect_ctx SEC(".maps");
+
+static __always_inline int enter_tcp_connect(struct pt_regs *ctx,
+					     struct sock *sk)
+{
+	__u64 pid_tgid = bpf_get_current_pid_tgid();
+	__u32 pid = pid_tgid >> 32;
+	__u32 tid = pid_tgid;
+	__u64 uid_gid = bpf_get_current_uid_gid();
+	__u32 uid = uid_gid;
+	__u64 mntns_id;
+
+	mntns_id = gadget_get_mntns_id();
+
+	if (filter_event(sk, uid, pid, mntns_id, connect))
+		return 0;
+
+	bpf_map_update_elem(&tcp_connect_ctx, &tid, &sk, 0);
+	return 0;
+}
+
+static __always_inline int exit_tcp_connect(struct pt_regs *ctx, int ret,
+					    __u16 family)
+{
+	__u64 pid_tgid = bpf_get_current_pid_tgid();
+	__u32 tid = pid_tgid;
+	struct tuple_key_t tuple = {};
+	struct gadget_process proc = {};
+	struct sock **skpp;
+	struct sock *sk;
+
+	skpp = bpf_map_lookup_elem(&tcp_connect_ctx, &tid);
+	if (!skpp)
+		return 0;
+
+	if (ret)
+		goto end;
+
+	sk = *skpp;
+
+	if (!fill_tuple(&tuple, sk, family))
+		goto end;
+
+	gadget_process_populate(&proc);
+	bpf_map_update_elem(&tuplepid, &tuple, &proc, 0);
+
+end:
+	bpf_map_delete_elem(&tcp_connect_ctx, &tid);
+	return 0;
+}
+
+SEC("kprobe/tcp_v4_connect")
+int BPF_KPROBE(ig_tcp_v4_co_e, struct sock *sk)
+{
+	return enter_tcp_connect(ctx, sk);
+}
+
+SEC("kretprobe/tcp_v4_connect")
+int BPF_KRETPROBE(ig_tcp_v4_co_x, int ret)
+{
+	return exit_tcp_connect(ctx, ret, AF_INET);
+}
+
+SEC("kprobe/tcp_v6_connect")
+int BPF_KPROBE(ig_tcp_v6_co_e, struct sock *sk)
+{
+	return enter_tcp_connect(ctx, sk);
+}
+
+SEC("kretprobe/tcp_v6_connect")
+int BPF_KRETPROBE(ig_tcp_v6_co_x, int ret)
+{
+	return exit_tcp_connect(ctx, ret, AF_INET6);
+}
```

Comment:
Then I would like to split them up again in two different files. With the different files it would also make things clearer which kprobes are used for which part of the tracing.
But we can't have multiple .c files yet and we dont want to include .c files

What should I do?